pkg/controllers/repositoryserver/secrets_manager.go

```diff
@@ -74,7 +74,11 @@ func (h *RepoServerHandler) getSecretsFromCR(ctx context.Context) error {
 	return nil
 }
 
-func (h *RepoServerHandler) fetchSecret(ctx context.Context, ref corev1.SecretReference) (*corev1.Secret, error) {
+func (h *RepoServerHandler) fetchSecret(ctx context.Context, ref *corev1.SecretReference) (*corev1.Secret, error) {
+	if ref == nil {
+		return nil, errors.New("repository server CR does not have a secret reference set")
```

Comment:
```suggestion
		return nil, errors.New("repository server CR does not have a TLS secret reference set")
```

?

Reply:
Its for fetch all the secrets, not just TLS secret.